pkg/engine/engine.go

```diff
@@ -595,17 +595,17 @@ func SupportsLineNumbers(sourceType sourcespb.SourceType) bool {
 
 // FragmentLineOffset sets the line number for a provided source chunk with a given detector result.
 func FragmentLineOffset(chunk *sources.Chunk, result *detectors.Result) (int64, bool) {
-	lines := bytes.Split(chunk.Data, []byte("\n"))
-	for i, line := range lines {
-		if bytes.Contains(line, result.Raw) {
-			// if the line contains the ignore tag, we should ignore the result
-			if bytes.Contains(line, []byte(ignoreTag)) {
-				return int64(i), true
-			}
-			return int64(i), false
-		}
+	before, after, found := bytes.Cut(chunk.Data, result.Raw)
+	if !found {
+		return 0, false
+	}
+	lineNumber := int64(len(bytes.Split(before, []byte("\n")))) - 1
```

Comment:
i'm not sure how big chunks can get, but if they can get big we could save a bunch of memory by simply counting `\n` characters (rather than splitting them), right?

Reply:
soo.. I had that exact same thought and tested it out. for some reason `Split` is way faster than counting:

```
BenchmarkSplitIt-8         14970             77778 ns/op          106525 B/op          1 allocs/op
BenchmarkIterateIt-8        9021            130181 ns/op              46 B/op          0 allocs/op
```

Reply:
I think in this case it makes sense to prioritize speed over memory consumption, though I could be persuaded otherwise.

It's not *exactly* in the hot path but close to it.

Reply:
oh, I'm dumb. there's a [bytes.Count](https://pkg.go.dev/bytes#Count) function that `Split` uses internally.